packages/side-nav/src/vaadin-side-nav-item.js

```diff
@@ -0,0 +1,131 @@
+/**
+ * @license
+ * Copyright (c) 2017 - 2023 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+import { html, LitElement } from 'lit';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { ElementMixin } from '@vaadin/component-base/src/element-mixin.js';
+import { PolylitMixin } from '@vaadin/component-base/src/polylit-mixin.js';
+import { ThemableMixin } from '@vaadin/vaadin-themable-mixin/vaadin-themable-mixin.js';
+import { sideNavItemStyles } from '../theme/lumo/vaadin-side-nav-styles.js';
+
+function isEnabled() {
+  return window.Vaadin && window.Vaadin.featureFlags && !!window.Vaadin.featureFlags.sideNavComponent;
+}
+
+class SideNavItem extends ElementMixin(ThemableMixin(PolylitMixin(LitElement))) {
+  static get is() {
+    return 'vaadin-side-nav-item';
+  }
+
+  static get styles() {
+    return sideNavItemStyles;
+  }
+
+  static get properties() {
+    return {
+      path: String,
+
+      expanded: {
+        type: Boolean,
+        value: false,
+        reflectToAttribute: true,
+      },
+
+      active: {
+        type: Boolean,
+        value: false,
+        reflectToAttribute: true,
+      },
+    };
+  }
+
+  get button() {
+    return this.shadowRoot.querySelector('button');
+  }
+
+  get childrenSlot() {
+    return this.shadowRoot.querySelector('#children');
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.setAttribute('role', 'listitem');
+    this._updateActive();
+    this.__boundUpdateActive = this._updateActive.bind(this);
+    window.addEventListener('popstate', this.__boundUpdateActive);
+  }
+
+  /** @protected */
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    window.removeEventListener('popstate', this.__boundUpdateActive);
+  }
+
+  /** @protected */
+  render() {
+    return html`
+      <a href="${ifDefined(this.path)}" part="item" aria-current="${this.active ? 'page' : false}">
+        <slot name="prefix"></slot>
+        <slot></slot>
+        <slot name="suffix"></slot>
+        <button
+          part="toggle-button"
+          @click="${this.toggleExpanded}"
+          ?hidden="${!this.querySelector('[slot=children]')}"
+          aria-controls="children"
+          aria-expanded="${this.expanded}"
+          aria-label="Toggle child items"
```

Comment:
This needs i18n.. could be done on side-nav level?

Reply:
Yes, it is on our roadmap for the stable version of the component.